pkgs/misc/emulators/rpcs3/default.nix

```diff
@@ -1,23 +1,26 @@
-{ mkDerivation, lib, fetchgit, cmake, pkg-config, git
+{ mkDerivation, lib, fetchFromGitHub, cmake, pkg-config, git
 , qtbase, qtquickcontrols, openal, glew, vulkan-headers, vulkan-loader, libpng
-, ffmpeg, libevdev, python3
+, ffmpeg, libevdev, curl, /* wolfssl, */ python3
+, sdl2Support ? true, SDL2
 , pulseaudioSupport ? true, libpulseaudio
 , waylandSupport ? true, wayland
 , alsaSupport ? true, alsaLib
 }:
 
 let
-  majorVersion = "0.0.12";
-  gitVersion = "10811-a86a3d2fe"; # echo $(git rev-list HEAD --count)-$(git rev-parse --short HEAD)
+  majorVersion = "0.0.16";
+  gitVersion = "12235-a4f4b81e6"; # echo $(git rev-list HEAD --count)-$(git rev-parse --short HEAD)
 in
 mkDerivation {
   pname = "rpcs3";
   version = "${majorVersion}-${gitVersion}";
 
-  src = fetchgit {
-    url = "https://github.com/RPCS3/rpcs3";
-    rev = "v${majorVersion}";
-    sha256 = "182rkmbnnlcfzam4bwas7lwv10vqiqvvaw3299a3hariacd7rq8x";
+  src = fetchFromGitHub {
+    owner = "RPCS3";
+    repo = "rpcs3";
+    rev = "a4f4b81e6b0c00f4c30f9f5f182e5fe56f9fb03c";
+    fetchSubmodules = true;
```

Comment:
Please devendor all their sub modules. We do not want to accidentally link against an old curl, ffmpeg and many others.

Reply:
Not possible without upstream changes. Most of this is handled by the `-DUSE_SYSTEM_*` options (except WolfSSL, obviously), however there's others for which there's no override. The build fails without `fetchSubmodules = true;`

Reply:
I am mostly concerned about curl and ffmpeg which should be updated regularly.

Reply:
Both curl and ffmpeg are de-vendored, see:
 ```nix
    "-DUSE_SYSTEM_FFMPEG=ON"
    "-DUSE_SYSTEM_CURL=ON"
```